src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -90,6 +90,11 @@ void Terminal::UpdateSettings(ICoreSettings settings)
     _autoMarkPrompts = settings.AutoMarkPrompts();
     _rainbowSuggestions = settings.RainbowSuggestions();
 
+    if (_stateMachine)
+    {
+        SetVtChecksumReportSupport(settings.AllowVtChecksumReport());

```

Comment:
what if `stateMachine` comes along later but there's no UpdateSettings call?

Reply:
The only place we set `_stateMachine` is in `Terminal::Create()`. I went ahead and set a breakpoint there and on `Terminal::UpdateSettings()`. In practice, when a Terminal is created, we hit `UpdateSettings()` --> `Create()` --> `UpdateSettings()`.

If you're curious, it's because `CreateFromSettings()` calls `Create()` (which sets the `_stateMachine`) and then `UpdateSettings()`.

So, really that check is just for that first call when we initialize the `ControlCore` (which ends on an `UpdateSettings()`). So I think we're good here.

Reply:
woot, thanks